core/services.go

```diff
@@ -362,24 +365,211 @@ func (app *Application) uploadProfileImageToAws(image image.Image, filename stri
 	return nil, nil
 }
 
-func (app *Application) getTwitterPosts(userID string, twitterQueryParams string, force bool) (map[string]interface{}, error) {
+func (s *servicesImpl) GetTwitterPosts(userID string, twitterQueryParams string, force bool) (map[string]interface{}, error) {
 	var err error
-	posts := app.cacheAdapter.GetTwitterPosts(userID, twitterQueryParams)
+	posts := s.app.cacheAdapter.GetTwitterPosts(userID, twitterQueryParams)
 	if posts == nil || force {
-		app.cacheLock.Lock()
-		posts = app.cacheAdapter.GetTwitterPosts(userID, twitterQueryParams)
+		s.app.cacheLock.Lock()
+		posts = s.app.cacheAdapter.GetTwitterPosts(userID, twitterQueryParams)
 		if posts == nil || force {
 			if force {
-				app.cacheAdapter.ClearTwitterCacheForUser(userID)
+				s.app.cacheAdapter.ClearTwitterCacheForUser(userID)
 			}
-			posts, err = app.twitterAdapter.GetTwitterPosts(userID, twitterQueryParams)
+			posts, err = s.app.twitterAdapter.GetTwitterPosts(userID, twitterQueryParams)
 			if err == nil {
-				app.cacheAdapter.SetTwitterPosts(userID, twitterQueryParams, posts)
+				s.app.cacheAdapter.SetTwitterPosts(userID, twitterQueryParams, posts)
 			} else {
 				fmt.Printf("error feeding twitter: %s", err)
 			}
 		}
-		app.cacheLock.Unlock()
+		s.app.cacheLock.Unlock()
 	}
 	return posts, err
 }
+
+func (s *servicesImpl) GetDataContentItem(claims *tokenauth.Claims, key string) (*model.DataContentItem, error) {
+	item, err := s.app.storage.FindDataContentItem(&claims.AppID, claims.OrgID, key)
+	if err != nil {
+		return nil, err
+	}
+	return item, nil
+}
+
+func (s *servicesImpl) GetDataContentItems(claims *tokenauth.Claims, category string) ([]*model.DataContentItem, error) {
+	item, err := s.app.storage.FindDataContentItems(&claims.AppID, claims.OrgID, category)
+	if err != nil {
+		return nil, err
+	}
+	return item, nil
+}
+
+func (s *servicesImpl) CreateDataContentItem(claims *tokenauth.Claims, item *model.DataContentItem) (*model.DataContentItem, error) {
+	item.ID = uuid.NewString()
+	item.AppID = &claims.AppID
+	item.OrgID = claims.OrgID
+	item.DateCreated = time.Now().UTC()
+	item, err := s.app.storage.CreateDataContentItem(item)
```

Comment:
I think we should check the category permissions against the token permissions to make sure the user can create the content item.